network/nxos/nxos_acl_interface.py

```diff
@@ -0,0 +1,537 @@
+        if name:
+            temp['name'] = name
+
+        if temp:
+            acl_list.append(temp)
+
+    existing_no_null = []
+    for each in acl_list:
+        if each.get('name') == acl:
+            interfaces = each.get('interfaces')
+            for interface in interfaces:
+                new_temp = {}
+                new_temp['name'] = acl
+                new_temp.update(interface)
+                existing_no_null.append(new_temp)
+    return existing_no_null
+
+
+def other_existing_acl(get_existing, interface, direction):
+    # now we'll just get the interface in question
+    # needs to be a list since same acl could be applied in both dirs
+    acls_interface = []
+    if get_existing:
+        for each in get_existing:
+            if each.get('interface').lower() == interface:
+                acls_interface.append(each)
+    else:
+        acls_interface = []
+
+    if acls_interface:
+        this = {}
+        for each in acls_interface:
+            if each.get('direction') == direction:
+                this = each
+    else:
+        acls_interface = []
+        this = {}
+
+    return acls_interface, this
+
+
+def apply_acl(proposed):
+    commands = []
+
+    commands.append('interface ' + proposed.get('interface'))
+    direction = proposed.get('direction')
+    if direction == 'egress':
+        cmd = 'ip access-group {0} {1}'.format(proposed.get('name'), 'out')
+    elif direction == 'ingress':
+        cmd = 'ip access-group {0} {1}'.format(proposed.get('name'), 'in')
+    commands.append(cmd)
+
+    return commands
+
+
+def remove_acl(proposed):
+    commands = []
+
+    commands.append('interface ' + proposed.get('interface'))
+    direction = proposed.get('direction')
+    if direction == 'egress':
+        cmd = 'no ip access-group {0} {1}'.format(proposed.get('name'), 'out')
+    elif direction == 'ingress':
+        cmd = 'no ip access-group {0} {1}'.format(proposed.get('name'), 'in')
+    commands.append(cmd)
+
+    return commands
+
+
+def flatten_list(command_lists):
+    flat_command_list = []
+    for command in command_lists:
+        if isinstance(command, list):
+            flat_command_list.extend(command)
+        else:
+            flat_command_list.append(command)
+    return flat_command_list
+
+
+def execute_config_command(commands, module):
+    try:
+        module.configure(commands)
+    except ShellError:
+        clie = get_exception()
+        module.fail_json(msg='Error sending CLI commands',
+                         error=str(clie), commands=commands)
+    except AttributeError:
+        try:
+            commands.insert(0, 'configure')
+            module.cli.add_commands(commands, output='config')
+            module.cli.run_commands()
+        except ShellError:
+            clie = get_exception()
+            module.fail_json(msg='Error sending CLI commands',
+                             error=str(clie), commands=commands)
+
+
+def main():
+    argument_spec = dict(
```

Comment:
As stated in other reviews, please split `argument_spec` so only the specific options defined in the module are listed here.